.github/workflows/build.yml

```diff
@@ -0,0 +1,80 @@
+name: Build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  build:
+    name: Build library
+    runs-on: macos-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: 11
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Build project
+        run: ./gradlew build -x detekt
+
+  run_tests:
+    name: Run tests
+    runs-on: macos-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: 11
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Unit tests
+        run: ./gradlew test -x detekt
```

Comment:
I think that it would be nice to have coverage information somehow available. 
Tests could also fail if the coverage is below some threshold.

Reply:
done
is generating raport from coverage necessary? idk why but "Upload coverage report" job is cancelled

Reply:
It would be nice to have this information

Reply:
ok done